backend/file.go

```diff
@@ -0,0 +1,237 @@
+package backend
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"golang.org/x/crypto/argon2"
+	"golang.org/x/crypto/chacha20poly1305"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"time"
+)
+
+type FileStore struct {
+	storeLocation string
+	projectID     string
+	cryptoKey     []byte
+}
+
+func NewFileStore(storeLocation, projectID, cryptoKey string) *FileStore {
+	disc := &FileStore{
+		projectID: projectID,
+		cryptoKey: []byte(cryptoKey),
+	}
+	disc.storeLocation = disc.secretStoreLocation(storeLocation, projectID)
+	return disc
+}
+
+type FileStoreEntry struct {
+	Value   []byte
+	KeyInfo Key
+}
+
+// Get reads the store from file, fetches and decrypt the value for given key
+func (d *FileStore) Get(_ context.Context, _ *Profile, key string) ([]byte, error) {
+	storeData, err := d.getStore()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, data := range storeData {
+		if data.KeyInfo.Name == key {
+			data, err := d.decrypt(data.Value, d.cryptoKey)
+			if err != nil {
+				return nil, fmt.Errorf("message authentication failed")
+			}
+			return data, nil
+		}
+	}
+	return nil, fmt.Errorf("%s not found", key)
+}
+
+// List reads the store from file, and fetch all keys
+func (d *FileStore) List(_ context.Context, _ *Profile) ([]Key, error) {
+	storeData, err := d.getStore()
+	if err != nil {
+		return nil, err
+	}
+	var keys []Key
```

Comment:
[tip] you can also name the return values so that you can skip the var declaration